kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/ClientConsumerRecord.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+/**
+ * The type Client consumer record.
+ */
+public class ClientConsumerRecord extends BaseConsumerRecord {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientConsumerRecord.class);
+
+    private long timestamp;
+    private String timestampType;
+    private List<Map<String, String>> headers;
```

Comment:
List<Map<String, String>> look really surprising. Might it be you mean List<Entry<String, String>>?

Reply:
For `strimzi_test_client` I can change it to `Entry` instead of a `Map` because it is reported like this:
```
headers:[{"kroxylicious/encryption": "value"}]
```
But `kaf` needs to keep it as a `Map` because it is reported like this:
```
"headers":[{"Key":"a3JveHlsaWNpb3VzLmlvL2VuY3J5cHRpb24=","Value":"Ag=="}]
```

Reply:
You should be able to hide that detail in the concrete implement.

Have ConsumerRecord interface declare:

Map<String, String> headers();

then in the concrete implementation, the constructors look like:

public StrimziTestClientConsumerRecord(List<Entry<String, String>> headers, other fields)

public KafClientConsumerRecord(Map<Map<String, String>> headers, other fields)

within the constructor of each, map from the data-structure provided by the client to the normalised form Map<String, String>.

Reply:
Thanks! I didn't know about `@JsonCreator`